arrays/arrays.c

```diff
@@ -19,45 +19,51 @@ typedef struct Array {
 /*****
  * Allocate memory for a new array
  *****/
-Array *create_array (int capacity) {
-  // Allocate memory for the Array struct
+Array *create_array(int capacity)
+{
+    // Allocate memory for the Array struct
+    Array *newArr = malloc(sizeof(Array));
 
-  // Set initial values for capacity and count
+    // Set initial values for capacity and count
+    newArr->capacity = capacity;
+    newArr->count = 0;
 
-  // Allocate memory for elements
+    // Allocate memory for elements
+    newArr->elements = malloc(capacity * sizeof(char));
 
+    return newArr;
 }
 
-
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
-
-  // Free all elements
+void destroy_array(Array *arr)
+{
 
-  // Free array
+    // Free all elements
+    // Only one of the elements is a pointer to something we manually allocated memory for, so here it is
```

Comment:
veeeery true! unless of course you decide to implement `insert` and `append` like we did during lecture/in the solution code.
lecture's/solution code's `insert` and `append` methods don't point directly to the element added, but instead use `strdup()` to make a copy of that element being added. since `strdup()` malloc's under the hood, you'd need to free those as well.